RestCPSubsystemTest.test_removeCPMemberFromNonMaster

Failed on Sonar build : http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-3.x-sonar/37/testReport/com.hazelcast.internal.ascii/RestCPSubsystemTest/test_removeCPMemberFromNonMaster/

**Stacktrace**

```
java.lang.NullPointerException
	at com.hazelcast.internal.ascii.RestCPSubsystemTest.test_removeCPMemberFromNonMaster(RestCPSubsystemTest.java:413)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

Standard output : 
```
3:20:57,279 ERROR |test_removeCPMemberFromNonMaster| - [MetadataRaftGroupManager] hz._hzInstance_11473_dev.cached.thread-2 - [127.0.0.1]:5702 [dev] [3.12-SNAPSHOT] Could not initialize METADATA CP group with CP members: [CPMember{uuid=1eb4090f-5c3c-4a4b-acac-7266d0905993, address=[127.0.0.1]:5703}, CPMember{uuid=4405f35d-53c8-46b0-a0ae-b02cad6fc82f, address=[127.0.0.1]:5702}, CPMember{uuid=e8f166e2-f05a-465a-b0a7-a15caa43ba73, address=[127.0.0.1]:5701}]
java.util.concurrent.ExecutionException: com.hazelcast.core.HazelcastInstanceNotActiveException: Hazelcast instance is not active!
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:92) ~[classes/:?]
	at com.hazelcast.spi.impl.AbstractInvocationFuture.get(AbstractInvocationFuture.java:162) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.get(InvocationFuture.java:177) ~[classes/:?]
	at com.hazelcast.cp.internal.MetadataRaftGroupManager$DiscoverInitialCPMembersTask.commitMetadataRaftGroupInit(MetadataRaftGroupManager.java:1106) [classes/:?]
	at com.hazelcast.cp.internal.MetadataRaftGroupManager$DiscoverInitialCPMembersTask.run(MetadataRaftGroupManager.java:1030) [classes/:?]
	at com.hazelcast.util.executor.CachedExecutorServiceDelegate$Worker.run(CachedExecutorServiceDelegate.java:227) [classes/:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_201]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_201]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_201]
	at com.hazelcast.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:64) [classes/:?]
	at com.hazelcast.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:80) [classes/:?]
Caused by: com.hazelcast.core.HazelcastInstanceNotActiveException: Hazelcast instance is not active!
	at com.hazelcast.spi.impl.operationservice.impl.InvocationRegistry.shutdown(InvocationRegistry.java:198) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.OperationServiceImpl.shutdownInvocations(OperationServiceImpl.java:503) ~[classes/:?]
	at com.hazelcast.spi.impl.NodeEngineImpl.shutdown(NodeEngineImpl.java:478) ~[classes/:?]
	at com.hazelcast.instance.Node.shutdownServices(Node.java:567) ~[classes/:?]
	at com.hazelcast.instance.Node.shutdown(Node.java:498) ~[classes/:?]
	at com.hazelcast.instance.LifecycleServiceImpl.shutdown(LifecycleServiceImpl.java:98) ~[classes/:?]
	at com.hazelcast.instance.LifecycleServiceImpl.terminate(LifecycleServiceImpl.java:86) ~[classes/:?]
	at com.hazelcast.internal.ascii.RestCPSubsystemTest.test_removeCPMemberFromNonMaster(RestCPSubsystemTest.java:411) ~[test-classes/:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_201]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_201]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_201]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_201]
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12) ~[junit-4.12.jar:4.12]
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47) ~[junit-4.12.jar:4.12]
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17) ~[junit-4.12.jar:4.12]
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:106) ~[test-classes/:?]
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:98) ~[test-classes/:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[?:1.8.0_201]
	at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_201]
	at ------ submitted from ------.(Unknown Source) ~[?:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolve(InvocationFuture.java:126) ~[classes/:?]
	at com.hazelcast.spi.impl.operationservice.impl.InvocationFuture.resolveAndThrowIfException(InvocationFuture.java:79) ~[classes/:?]
	... 10 more
```

--- a/hazelcast/src/test/java/com/hazelcast/cp/internal/CPMemberAddRemoveTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/cp/internal/CPMemberAddRemoveTest.java
@@ -652,12 +652,7 @@ public class CPMemberAddRemoveTest extends HazelcastRaftTestSupport {
         }, 5);
 
         instances[0].getCPSubsystem().getCPSubsystemManagementService().promoteToCPMember().get();
-        assertTrueEventually(new AssertTask() {
-            @Override
-            public void run() {
-                assertNotNull(instances[0].getCPSubsystem().getLocalCPMember());
-            }
-        });
+        assertNotNull(instances[0].getCPSubsystem().getLocalCPMember());
         assertNotEquals(localCpMember, instances[0].getCPSubsystem().getLocalCPMember());
     }
 

--- a/hazelcast/src/test/java/com/hazelcast/internal/ascii/RestCPSubsystemTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/ascii/RestCPSubsystemTest.java
@@ -35,6 +35,7 @@ import com.hazelcast.internal.json.JsonArray;
 import com.hazelcast.internal.json.JsonObject;
 import com.hazelcast.internal.json.JsonValue;
 import com.hazelcast.spi.properties.GroupProperty;
+import com.hazelcast.test.AssertTask;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.annotation.SlowTest;
@@ -54,6 +55,7 @@ import java.util.concurrent.ExecutionException;
 import static com.hazelcast.cp.CPGroup.DEFAULT_GROUP_NAME;
 import static com.hazelcast.cp.CPGroup.METADATA_CP_GROUP_NAME;
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
 @RunWith(HazelcastSerialClassRunner.class)
@@ -124,9 +126,18 @@ public class RestCPSubsystemTest extends HazelcastTestSupport {
 
     @Test
     public void test_getMetadataCPGroupByName() throws IOException {
-        HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertNotNull(instance1.getCPSubsystem().getLocalCPMember());
+                assertNotNull(instance2.getCPSubsystem().getLocalCPMember());
+                assertNotNull(instance3.getCPSubsystem().getLocalCPMember());
+            }
+        });
 
         HTTPCommunicator communicator = new HTTPCommunicator(instance1);
 
@@ -161,9 +172,18 @@ public class RestCPSubsystemTest extends HazelcastTestSupport {
 
     @Test
     public void test_getDefaultCPGroupByName() throws IOException {
-        HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertNotNull(instance1.getCPSubsystem().getLocalCPMember());
+                assertNotNull(instance2.getCPSubsystem().getLocalCPMember());
+                assertNotNull(instance3.getCPSubsystem().getLocalCPMember());
+            }
+        });
 
         instance1.getCPSubsystem().getAtomicLong("long1").set(5);
 
@@ -200,9 +220,18 @@ public class RestCPSubsystemTest extends HazelcastTestSupport {
 
     @Test
     public void test_getCustomCPGroupByName() throws IOException {
-        HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertNotNull(instance1.getCPSubsystem().getLocalCPMember());
+                assertNotNull(instance2.getCPSubsystem().getLocalCPMember());
+                assertNotNull(instance3.getCPSubsystem().getLocalCPMember());
+            }
+        });
 
         instance1.getCPSubsystem().getAtomicLong("long1@custom").set(5);
 
@@ -252,10 +281,19 @@ public class RestCPSubsystemTest extends HazelcastTestSupport {
 
     @Test
     public void test_getLocalCPMember() throws IOException {
-        HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
         HazelcastInstance instance4 = Hazelcast.newHazelcastInstance(config);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertNotNull(instance1.getCPSubsystem().getLocalCPMember());
+                assertNotNull(instance2.getCPSubsystem().getLocalCPMember());
+                assertNotNull(instance3.getCPSubsystem().getLocalCPMember());
+            }
+        });
 
         ConnectionResponse response1 = new HTTPCommunicator(instance1).getLocalCPMember();
         ConnectionResponse response2 = new HTTPCommunicator(instance2).getLocalCPMember();
@@ -278,9 +316,18 @@ public class RestCPSubsystemTest extends HazelcastTestSupport {
 
     @Test
     public void test_getCPMembers() throws IOException {
-        HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertNotNull(instance1.getCPSubsystem().getLocalCPMember());
+                assertNotNull(instance2.getCPSubsystem().getLocalCPMember());
+                assertNotNull(instance3.getCPSubsystem().getLocalCPMember());
+            }
+        });
 
         ConnectionResponse response = new HTTPCommunicator(instance1).getCPMembers();
 
@@ -360,9 +407,16 @@ public class RestCPSubsystemTest extends HazelcastTestSupport {
 
     @Test
     public void test_removeCPMember() throws IOException {
-        HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertNotNull(instance3.getCPSubsystem().getLocalCPMember());
+            }
+        });
 
         CPMember crashedCPMember = instance3.getCPSubsystem().getLocalCPMember();
         instance3.getLifecycleService().terminate();
@@ -378,7 +432,14 @@ public class RestCPSubsystemTest extends HazelcastTestSupport {
     public void test_removeCPMember_withInvalidCredentials() throws IOException {
         HazelcastInstance instance1 = Hazelcast.newHazelcastInstance(config);
         HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertNotNull(instance3.getCPSubsystem().getLocalCPMember());
+            }
+        });
 
         CPMember crashedCPMember = instance3.getCPSubsystem().getLocalCPMember();
         instance3.getLifecycleService().terminate();
@@ -404,8 +465,15 @@ public class RestCPSubsystemTest extends HazelcastTestSupport {
     @Test
     public void test_removeCPMemberFromNonMaster() throws IOException {
         Hazelcast.newHazelcastInstance(config);
-        HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
+        final HazelcastInstance instance2 = Hazelcast.newHazelcastInstance(config);
         HazelcastInstance instance3 = Hazelcast.newHazelcastInstance(config);
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() {
+                assertNotNull(instance2.getCPSubsystem().getLocalCPMember());
+            }
+        });
 
         CPMember crashedCPMember = instance2.getCPSubsystem().getLocalCPMember();
         instance2.getLifecycleService().terminate();

